src/icrar/leap-accelerate-cli/main.cc

```diff
@@ -89,20 +89,23 @@ int main(int argc, char** argv)
 #else
     boost::optional<std::string> readAutocorrelations;
     app.add_option("-a,--autocorrelations", readAutocorrelations, "Set to true if measurement set rows store autocorrelations");
-    rawArgs.readAutocorrelations = readAutocorrelations.is_initialized() ? std::stoi(readAutocorrelations.get()) : (boost::optional<int>)boost::none;
 
     boost::optional<std::string> minimumBaselineThreshold;
     app.add_option("-m,--minimumBaselineThreshold", minimumBaselineThreshold, "Minimum baseline length in meters");
-    rawArgs.minimumBaselineThreshold = minimumBaselineThreshold.is_initialized() ? std::stod(minimumBaselineThreshold.get()) : (boost::optional<double>)boost::none;
 
     boost::optional<std::string> verbosity;
     app.add_option("-v,--verbosity", verbosity, "Verbosity (0=fatal, 1=error, 2=warn, 3=info, 4=debug, 5=trace), defaults to info");
-    rawArgs.verbosity = verbosity.is_initialized() ? boost::lexical_cast<bool>(verbosity.get()) : (boost::optional<bool>)boost::none;
 #endif
 
     try
     {
         app.parse(argc, argv);
+
+#if !__has_include(<optional>)
+        rawArgs.readAutocorrelations = readAutocorrelations.is_initialized() ? std::stoi(readAutocorrelations.get()) : (boost::optional<int>)boost::none;
+        rawArgs.minimumBaselineThreshold = minimumBaselineThreshold.is_initialized() ? std::stod(minimumBaselineThreshold.get()) : (boost::optional<double>)boost::none;
+        rawArgs.verbosity = verbosity.is_initialized() ? boost::lexical_cast<bool>(verbosity.get()) : (boost::optional<bool>)boost::none;
```

Comment:
It still appears to me the casts in this last line are to `int`, not to `bool`. There is a call to `std::stoi` above instead of a `boost::lexical_cast`, and you are already using `stod` too, so I'd make this one a `stoi` too to keep things aligned.

Also in these three cases the `rawArgs.*` members should be set to `boost::none` already when `rawArgs` gets constructed, so there's no need to set them to `none` again. So I'd go for something like (which also looks more like what you've done in the rest of the code, IIRC):

```c++
if (readAutocorrelations.is_initialized())
{
     rawArgs.readAutocorrelations = std::stoi(readAutocorrelations.get());
}
if (minimumBaselineThreshold.is_initialized())
{
    rawArgs.minimumBaselineThreshold = std::stod(minimumBaselineThreshold.get());
}
if (verbosity.is_initialized())
{
    rawArgs.verbosity = std::stoi(verbosity.get());
}
```